### PR TITLE
Add page.on('metric') type definition

### PR DESCRIPTION
**Pull request by ankur22**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/grafana/xk6-browser/issues/371, https://github.com/grafana/xk6-browser/issues/1487
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

File: types/k6/browser/index.d.ts [hunk 1]
```diff
@@ -936,6 +936,69 @@ export interface ConsoleMessage {
...
+export interface MetricMessage {
+    /**
+     * tag will match the given `tagMatch.matches` with the current metric's url
+     * and name tags. When a match is found it will use `tagMatch.name` to replace
+     * the existing url and name tag values.
+     * 
+     * Doing this will help group metrics with disparate url and name tags, which
+     * are in fact referencing the same resource, so that a correlation can be found
+     * over time and preventing high cardinality issues in the backend database that
+     * is storing the metrics.
```
Comment (inancgumus) [inline types/k6/browser/index.d.ts:966]: I would rephrase this to remove the explanation about k6 backend internals:

```suggestion
     * Doing this helps to group metrics with disparate url and name tags, which
     * are, in fact, referencing the same resource so that a correlation can be 
     * found over time to reduce the high cardinality of the metrics.
```

**Review by inancgumus — APPROVED** [review]
LGTM besides one point 🚀

File: types/k6/browser/index.d.ts (hunk 1 above)
Comment (ankur22) [reply types/k6/browser/index.d.ts]: Thanks for the suggestion! I used that and changed it a bit to:

```suggestion
     * Doing this helps to group metrics with disparate url and name tags, which
     * are, in fact, referencing the same resource so that a correlation can be 
     * found over time and to reduce the cardinality of the metrics.
```

e10c6f3a6ddbb72fb2bdfdeb7d621ad1cb3f8825
